tools/pythonpkg/tests/fast/pandas/test_df_object_resolution.py

```diff
@@ -424,6 +425,39 @@ def test_analyze_sample_too_small(self):
         with pytest.raises(duckdb.InvalidInputException, match="Failed to cast value: Unimplemented type for cast"):
             res = duckdb.query_df(x, "x", "select * from x").df()
 
+    def test_numeric_decimal_zero_fractional(self):
+        duckdb_conn = duckdb.connect()
+        decimals = pd.DataFrame(
+            data={
+                "0": [
+                    Decimal("0.00"),
+                    Decimal("125.90"),
+                    Decimal("0.001"),
+                    Decimal("2502.63"),
+                    Decimal("0.000123"),
+                    Decimal("0.00"),
+                    Decimal("321.00"),
+                ]
+            }
+        )
+        reference_query = """
+            CREATE TABLE tbl AS SELECT * FROM (
+                VALUES
+                (0.00),
+                (125.90),
+                (0.001),
+                (2502.63),
+                (0.000123),
+                (0.00),
+                (321.00)
+            ) tbl(a);
+        """
+        duckdb_conn.execute(reference_query)
+        reference = duckdb.query("select * from tbl", connection=duckdb_conn).fetchall()
+        conversion = duckdb.query_df(decimals, "x", "select * from x").fetchall()
+
```

Comment:
Could we also add some tests with more extreme decimal values? e.g.:

* Values that are individually in-range, but out of range when combined: `(0.0000000000000000000000000001), (10000000000000000000000000.0)`
* Mix of very different widths and scales: `(1234.0), (123456789.0), (1234567890123456789.0), (0.1234567890123456789)`
* Values that are out of range entirely: `(1.234567890123456789012345678901234567890), (123456789012345678901234567890123456789.0)`